File: src/main/java/club/kosya/duraexec/spring/BeansConfig.java
package club.kosya.duraexec.spring;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

@Configuration
public class BeansConfig {
    @Bean
    ExecutorService taskExecutor() {
        return Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());
    }
}

File: src/main/java/club/kosya/duraexec/spring/BeansConfig.kt
package club.kosya.duraexec.spring

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors

@Configuration
class BeansConfig {
    @Bean
    fun taskExecutor(): ExecutorService = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors())
}
